sdgx/data_models/metadata.py

```diff
@@ -70,6 +93,159 @@ class Metadata(BaseModel):
+            pii = column_kwargs.get("pii", True)
+            column_kwargs["pii"] = pii
+
+        self.columns[column_name] = column_kwargs
+
+    def update_column(self, column_name, **kwargs):
+        """Update an existing column in the ``SingleTableMetadata``.
+
+        Args:
+            column_name (str):
+                The column name to be updated.
+            **kwargs (type):
+                Any key word arguments that describe metadata for the column.
+
+        Raises:
+            - ``InvalidMetadataError`` if the column doesn't already exist in the
+              ``SingleTableMetadata``.
+            - ``InvalidMetadataError`` if the column has unexpected values or ``kwargs`` for the
+              current
+              ``sdtype``.
+            - ``InvalidMetadataError`` if the ``pii`` value is not ``True`` or ``False`` when
+               present.
+        """
+        _kwargs = deepcopy(kwargs)
+        if "sdtype" in kwargs:
+            sdtype = kwargs.pop("sdtype")
+        else:
+            sdtype = self.columns[column_name]["sdtype"]
+            _kwargs["sdtype"] = sdtype
+
+        self.columns[column_name] = _kwargs
+
+    def set_primary_key(self, column_name):
+        """Set the metadata primary key.
+
+        Args:
+            column_name (str):
+                Name of the primary key column(s).
+        """
+        if column_name in self.alternate_keys:
+            warnings.warn(
+                f"'{column_name}' is currently set as an alternate key and will be removed from "
+                "that list."
+            )
+            self.alternate_keys.remove(column_name)
+
+        if self.primary_key is not None:
+            warnings.warn(
+                f"There is an existing primary key '{self.primary_key}'."
+                " This key will be removed."
+            )
+
+        self.primary_key = column_name
+
+    def remove_primary_key(self):
+        """Remove the metadata primary key."""
+        if self.primary_key is None:
+            warnings.warn("No primary key exists to remove.")
+
+        self.primary_key = None
+
+    def add_column_relationship(self, relationship_type, column_names):
+        """Add a column relationship to the metadata.
+
+        Args:
+            relationship_type (str):
+                Type of column relationship.
+            column_names (list[str]):
+                List of column names in the relationship.
+        """
+        relationship = {"type": relationship_type, "column_names": column_names}
+        to_check = [relationship] + self.column_relationships
+        self.column_relationships.append(relationship)
+
+    def _get_primary_and_alternate_keys(self):
+        """Get set of primary and alternate keys.
+
+        Returns:
+            set:
+                Set of keys.
+        """
+        keys = set(self.alternate_keys)
+        if self.primary_key:
+            keys.update({self.primary_key})
+
+        return keys
```

Comment:
Have you ever test this function? Not sure whether `primary_key` is a dict..If so, we may need to chang the name